esp32_s3_lcd_ev_board/src/bsp_sub_board.c

```diff
@@ -0,0 +1,377 @@
+            .y_max = BSP_LCD_SUB_BOARD_2_V_RES,
+            .rst_gpio_num = GPIO_NUM_NC,
+            .int_gpio_num = GPIO_NUM_NC,
+            .levels = {
+                .reset = 0,
+                .interrupt = 0,
+            },
+            .flags = {
+                .swap_xy = 0,
+                .mirror_x = 0,
+                .mirror_y = 0,
+            },
+        };
+        BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)BSP_I2C_NUM, &tp_io_config, &tp_io_handle));
+        BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_touch_new_i2c_ft5x06(tp_io_handle, &tp_cfg, &tp_handle));
+        break;
+    }
+    case SUB_BOARD_TYPE_3_800_480: {
+        const esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_GT1151_CONFIG();
+        const esp_lcd_touch_config_t tp_cfg = {
+            .x_max = BSP_LCD_SUB_BOARD_3_H_RES,
+            .y_max = BSP_LCD_SUB_BOARD_3_V_RES,
+            .rst_gpio_num = GPIO_NUM_NC,
+            .int_gpio_num = GPIO_NUM_NC,
+            .levels = {
+                .reset = 0,
+                .interrupt = 0,
+            },
+            .flags = {
+                .swap_xy = 0,
+                .mirror_x = 0,
+                .mirror_y = 0,
+            },
+        };
+
+        BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)BSP_I2C_NUM, &tp_io_config, &tp_io_handle));
+        BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_touch_new_i2c_gt1151(tp_io_handle, &tp_cfg, &tp_handle));
+        break;
+    }
+    default:
+        break;
+    }
+
+    if (ret_touch) {
+        *ret_touch = tp_handle;
+    }
+
+    return ESP_OK;
+}
+
+/**************************************************************************************************
+ *
+ * Other Function
+ *
+ **************************************************************************************************/
+uint16_t bsp_display_get_h_res(void)
+{
+    switch (sub_board_type) {
+    case SUB_BOARD_TYPE_2_480_480:
+        return BSP_LCD_SUB_BOARD_2_H_RES;
+    case SUB_BOARD_TYPE_3_800_480:
+        return BSP_LCD_SUB_BOARD_3_H_RES;
+    default:
+        ESP_LOGE(TAG, "Failed to get horizontal resolution, unknow sub-board");
+        return 0;
+    }
+}
+
+uint16_t bsp_display_get_v_res(void)
+{
+    switch (sub_board_type) {
+    case SUB_BOARD_TYPE_2_480_480:
+        return BSP_LCD_SUB_BOARD_2_V_RES;
+    case SUB_BOARD_TYPE_3_800_480:
+        return BSP_LCD_SUB_BOARD_3_V_RES;
+    default:
+        ESP_LOGE(TAG, "Failed to get vertical resolution, unknow sub-board");
+        return 0;
+    }
+}
+
+bsp_sub_board_type_t bsp_display_get_sub_board_type(void)
+{
+    return sub_board_type;
+}
+
+static esp_err_t detect_sub_board_type(void)
+{
+    if (sub_board_type != SUB_BOARD_TYPE_UNKNOW) {
+        return ESP_OK;
+    }
+
+    bsp_sub_board_type_t detect_type = SUB_BOARD_TYPE_UNKNOW;
+    BSP_ERROR_CHECK_RETURN_ERR(bsp_i2c_init());
+
+    i2c_cmd_handle_t cmd;
+    for (int i = 0; i < 0x7f; i++) {
```

Comment:
Try only `ESP_LCD_TOUCH_IO_I2C_FT5x06_ADDRESS` and `ESP_LCD_TOUCH_IO_I2C_GT1151_ADDRESS`.

Reply:
Sure, it's beterr.